### PR TITLE
Feat/1.2

### PR DESCRIPTION
**Pull request by Gevorg-Khachatryan-97**
- info modules for prism v3 modules
- vm operations

### CODE REVIEW

**Comment by Gevorg-Khachatryan-97** [comment]
ok-to-test

File: plugins/modules/ntnx_vms.py [hunk 1]
```diff
@@ -680,27 +729,22 @@
...
 from ..module_utils.prism.vms import VM  # noqa: E402
-from ..module_utils.utils import remove_param_with_none_value  # noqa: E402
 
 
 def get_module_spec():
     mutually_exclusive = [("name", "uuid")]
 
     entity_by_spec = dict(name=dict(type="str"), uuid=dict(type="str"))
 
-    network_spec = dict(
```
Comment (bhati-pradeep) [inline plugins/modules/ntnx_vms.py:694]: There are some params like network_spec, vcpus, etc are removed. Please check and restore them.

File: plugins/modules/ntnx_vms_clone.py [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+    task = Task(module)
+    task_uuid = result["task_uuid"]
+    resp = task.wait_for_completion(task_uuid, raise_error=raise_error)
+    result["response"] = resp
+    if not result.get("vm_uuid") and resp.get("entity_reference_list"):
+        result["vm_uuid"] = resp["entity_reference_list"][0]["uuid"]
+
+
+def run_module():
+    module = VMBaseModule(supports_check_mode=True)
```
Comment (bhati-pradeep) [inline plugins/modules/ntnx_vms_clone.py:306]: Confirm Once. We can override name and other params while cloning. Pleas check https://www.nutanix.dev/api_references/prism-central-v3/#/b3A6MjU1ODc4NDA-clone-a-vm

File: plugins/modules/ntnx_vms_info.py [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+
+    resp = vm.list(spec)
+
+    result["response"] = resp
+
+
+def run_module():
+    module = BaseInfoModule(
+        argument_spec=get_module_spec(),
+        supports_check_mode=True,
```
Comment (bhati-pradeep) [inline plugins/modules/ntnx_vms_info.py:291]: Info modules shouldn't have check modes

File: plugins/modules/ntnx_subnets_info.py [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+
+    resp = subnet.list(spec)
+
+    result["response"] = resp
+
+
+def run_module():
+    module = BaseInfoModule(
+        argument_spec=get_module_spec(),
+        supports_check_mode=True,
```
Comment (bhati-pradeep) [inline plugins/modules/ntnx_subnets_info.py:227]: Check mode are not required for info module

File: plugins/modules/ntnx_pbrs_info.py [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+
+    resp = pbr.list(spec)
+
+    result["response"] = resp
+
+
+def run_module():
+    module = BaseInfoModule(
+        argument_spec=get_module_spec(),
+        supports_check_mode=True,
```
Comment (bhati-pradeep) [inline plugins/modules/ntnx_pbrs_info.py:192]: Check mode are not required for info modules

File: plugins/modules/ntnx_floating_ips_info.py [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+
+    resp = floating_ip.list(spec)
+
+    result["response"] = resp
+
+
+def run_module():
+    module = BaseInfoModule(
+        argument_spec=get_module_spec(),
+        supports_check_mode=True,
```
Comment (bhati-pradeep) [inline plugins/modules/ntnx_floating_ips_info.py:169]: Check mode is not required for info modules

File: plugins/modules/ntnx_vpcs_info.py [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+
+    resp = vpc.list(spec)
+
+    result["response"] = resp
+
+
+def run_module():
+    module = BaseInfoModule(
+        argument_spec=get_module_spec(),
+        supports_check_mode=True,
```
Comment (bhati-pradeep) [inline plugins/modules/ntnx_vpcs_info.py:191]: check mode is not requried

File: plugins/modules/ntnx_ova.py [hunk 8]
```diff
@@ -0,0 +1,282 @@
...
+
+def run_module():
+    module = BaseModule(argument_spec=get_module_spec(), supports_check_mode=True)
+    result = {
+        "changed": False,
+        "error": None,
+        "response": None,
+        "vm_uuid": None,
+        "task_uuid": None,
+    }
```
Comment (bhati-pradeep) [inline plugins/modules/ntnx_ova.py:271]: remove_none_values

File: plugins/module_utils/base_info_module.py [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+class BaseInfoModule(BaseModule):
+    info_argument_spec = dict(
+        offset=dict(type="int"),
+        length=dict(type="int"),
+        filter=dict(type="dict"),
+        custom_filter=dict(type="dict"),
+    )
+
+    def __init__(self, **kwargs):
+        self.argument_spec = self.argument_spec.copy()
```
Comment (bhati-pradeep) [inline plugins/module_utils/base_info_module.py:19]: deepcopy()

File: plugins/module_utils/base_module.py [hunk 10]
```diff
@@ -38,10 +34,10 @@ class BaseModule(AnsibleModule):
     )
 
     def __init__(self, **kwargs):
+        argument_spec = self.argument_spec.copy()
```
Comment (bhati-pradeep) [inline plugins/module_utils/base_module.py:37]: deepcopy()

File: plugins/modules/ntnx_ova.py [hunk 11]
```diff
@@ -0,0 +1,282 @@
...
+    create(module, result)
```
[the remark this answers is not in the document]
Comment (bhati-pradeep) [reply plugins/modules/ntnx_ova.py:272]: Check : https://www.nutanix.dev/api_references/prism-central-v3/#/b3A6MjU1ODc1MzY-delete-a-existing-ova

File: tests/integration/targets/nutanix_vms/tasks/vm_operations.yml [hunk 12]
```diff
@@ -0,0 +1,206 @@
...
+  check_mode: yes
+
+- name: Creation Status
+  assert:
+    that:
+      - result.response is defined
+      - result.changed == false
+      - result.failed == false
+      - result.task_uuid != ""
+    success_msg: ' Success: returned  as expected '
```
Comment (bhati-pradeep) [inline tests/integration/targets/nutanix_vms/tasks/vm_operations.yml:53]: Is there any need for give trailing spaces ?

File: tests/integration/targets/nutanix_vms/vars/main.yml [hunk 13]
```diff
@@ -1,6 +1,6 @@
 ---
 centos: "CentOS-7-cloudinit" 
-ubuntu: "Ubuntu1404" 
+ubuntu: "Ubuntu-20.04.4"
```
Comment (bhati-pradeep) [inline tests/integration/targets/nutanix_vms/vars/main.yml:3]: make sure to add this image before running integration tests

**Review by bhati-pradeep — APPROVED** [review]
LGTM

**Comment by bhati-pradeep** [comment]
@Gevorg-Khachatryan-97 Please update the release version.

File: plugins/modules/ntnx_vms_clone.py (hunk 2 above)
Comment (Gevorg-Khachatryan-97) [reply plugins/modules/ntnx_vms_clone.py:306]: it's supporting

File: plugins/module_utils/entity.py [hunk 14]
```diff
@@ -249,7 +292,14 @@ def _build_url_with_query(self, url, query):
...
 
     def _fetch_url(
-        self, url, method, data=None, raise_error=True, no_response=False, timeout=30
+        self,
+        url,
+        method,
+        data=None,
+        raise_error=True,
+        no_response=False,
+        no_fail=False,
```
Comment (premkarat) [inline plugins/module_utils/entity.py:301]: check the use.

Comment (premkarat) [reply plugins/module_utils/entity.py:301]: use private functions

File: plugins/module_utils/entity.py [hunk 15]
```diff
@@ -165,7 +165,21 @@ def list(
         url = self.base_url if use_base_url else self.base_url + "/list"
         if endpoint:
             url = url + "/{0}".format(endpoint)
+
+        resp = self._fetch_url(
+            url,
+            method="POST",
+            data=data,
+            raise_error=False,
```
[the remark this answers is not in the document]
Comment (premkarat) [reply plugins/module_utils/entity.py:173]: ok first try to suppress error and then if it fails then get it part by part and raise error if it fails during this part fetch stage

**Review by premkarat — APPROVED** [review]

**Pull request merged**